csharp/ql/lib/semmle/code/csharp/frameworks/microsoft/AspNetCore.qll

```diff
@@ -185,7 +197,13 @@ class MicrosoftAspNetCoreMvcControllerBaseClass extends Class {
 /** A subtype of `Microsoft.AspNetCore.Mvc.Controller` or `Microsoft.AspNetCore.Mvc.ControllerBase`. */
 class MicrosoftAspNetCoreMvcController extends Class {
   MicrosoftAspNetCoreMvcController() {
-    this.getABaseType*() instanceof MicrosoftAspNetCoreMvcControllerBaseClass
+    (
+      this.getABaseType*() instanceof MicrosoftAspNetCoreMvcControllerBaseClass or
```

Comment:
If this should comply with the conjunction interpretation this should an "and" and then we need parenthesis around the disjunkt on l 202-203).
With the current implementation then "any" class where the name is postfixed with controller will satisfy l 202.

Reply:
I think this will have other side effects as well, if we perceive any class with a Controller suffix as being an ASP.NET Core controller:
In `Remote.qll` RemoteFlowSources are defined. With this extension it means that all parameters of any public method of a class suffixed with Controller will be considered a remote flow source (as far as I can tell). This will cause many false positives.

Reply:
Check if the project references asp.net? Please check the new commit if it is ok.